src/backend/looped/vehicle/navigation.cpp

```diff
@@ -22,27 +31,270 @@ namespace big
+					// only display notifications if setting is on
+					switch (direction)
+					{
+					case 1:
+						//g_notification_service->push_warning("Navigation", "Route Recalculation");
+						if (g.self.nav_text)
+							notify::above_map("Route Recalculation");
+						if (g.self.nav_voice)
+						{
+							audio::play_sound("RouteRecalc", 0);
+						}
+						break;
+					case 2:
+						//g_notification_service->push_warning("Navigation", "Please proceed to the highlighted route");
+						if (g.self.nav_text)
+							notify::above_map("Continue on Route");
+						if (g.self.nav_voice)
+						{
+							audio::play_sound("HighlightedRoute", 0);
+						}
+						break;
+					case 3:
+						switch (feetToAction)
+						{
+						case 130:
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("FeetUntilAction", 5280);
+								audio::play_sound("KeepLeft", 0);
+							}
+							break;
+
+						case 40:
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("FeetUntilAction", 1000);
+								audio::play_sound("KeepLeft", 0);
+							}
+							break;
+						case 25:
+							if (g.self.nav_text)
+								notify::above_map(std::format("Keep Left in {} feet", 500));
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("FeetUntilAction", 500);
+								audio::play_sound("KeepLeft", 0);
+							}
+							break;
+						case 15:
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("FeetUntilAction", 100);
+								audio::play_sound("KeepLeft", 0);
+							}
+							break;
+						case 10:
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("KeepLeft", 0);
+							}
+							break;
+						}
+						//g_notification_service->push_warning("Navigation", std::format("In {} feet, keep left", feetToAction));
+						break;
+					case 4:
+						switch (feetToAction)
+						{
+						case 130:
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("FeetUntilAction", 5280);
+								audio::play_sound("KeepRight", 0);
+							}
+							break;
+						case 40:
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("FeetUntilAction", 1000);
+								audio::play_sound("KeepRight", 0);
+							}
+							break;
+						case 25:
+							if (g.self.nav_text)
+								notify::above_map(std::format("Keep Right in {} feet", 500));
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("FeetUntilAction", 500);
+								audio::play_sound("KeepRight", 0);
+							}
+							break;
+						case 15:
+							audio::play_sound("FeetUntilAction", 100);
+							audio::play_sound("KeepRight", 0);
+							break;
+						case 10:
+							if (g.self.nav_voice)
+							{
+								audio::play_sound("KeepRight", 0);
+							}
+							break;
+						}
+						//g_notification_service->push_warning("Navigation", std::format("In {} feet, keep right", feetToAction));
+						break;
+					case 5:
+						switch (feetToAction)
+						{
+						case 130:
+							if (g.self.nav_voice)
```

Comment:
Why are you checking g.self.nav_voice a million times? This should either be checked at the head or returned if the user has this off.

Reply:
To know if the sound is enabled to play the sound if needed...

Reply:
did you even read what gir said or are you trolling?

Reply:
I am trolling lmao